x-pack/plugins/fleet/public/components/agent_enrollment_flyout/download_instructions.tsx

```diff
@@ -84,101 +69,63 @@ export const DownloadInstructions: React.FunctionComponent<Props> = ({
-      }}
+      defaultMessage="Copy or download the Kubernetes manifest. From the directory where the manifest is downloaded, run the apply command."
     />
-  ) : (
+  );
+
+  const k8sCopyYaml = (
+    <EuiCopy textToCopy={yaml}>
+      {(copy) => (
+        <EuiButton onClick={copy} iconType="copyClipboard">
+          <FormattedMessage
+            id="xpack.fleet.agentEnrollment.copyPolicyButton"
+            defaultMessage="Copy to clipboard"
+          />
+        </EuiButton>
+      )}
+    </EuiCopy>
+  );
+
+  const k8sYaml = (
+    <EuiCodeBlock language="yaml" style={{ maxHeight: 300 }} fontSize="m">
+      {yaml}
+    </EuiCodeBlock>
+  );
+
+  const downloadLink = core.http.basePath.prepend(
+    `${agentPolicyRouteService.getK8sFullDownloadPath()}?fleetServer=${fleetServer}&enrolToken=${enrollmentAPIKey}`
+  );
+
+  const downloadMsg = (
     <FormattedMessage
-      id="xpack.fleet.agentEnrollment.downloadDescription"
-      defaultMessage="Install the Elastic Agent on the hosts you wish to monitor. Do not install this agent policy on a host containing Fleet Server. You can download the Elastic Agent binaries and verification signatures from Elastic’s download page."
+      id="xpack.fleet.agentEnrollment.downloadManifestButtonk8s"
+      defaultMessage="Download Manifest"
     />
   );
 
-  const linuxUsers =
-    isK8s !== 'IS_KUBERNETES' ? (
-      <FormattedMessage
-        id="xpack.fleet.agentEnrollment.downloadUseLinuxInstaller"
-        defaultMessage="Linux users: We recommend the installer (TAR) over system packages (RPM/DEB) because it lets you upgrade your agent in Fleet."
-      />
-    ) : (
-      ''
-    );
-
-  const k8sCopyYaml =
-    isK8s === 'IS_KUBERNETES' ? (
-      <EuiCopy textToCopy={yaml}>
-        {(copy) => (
-          <EuiButton onClick={copy} iconType="copyClipboard">
-            <FormattedMessage
-              id="xpack.fleet.agentEnrollment.copyPolicyButton"
-              defaultMessage="Copy to clipboard"
-            />
-          </EuiButton>
-        )}
-      </EuiCopy>
-    ) : (
-      ''
-    );
-
-  const k8sYaml =
-    isK8s === 'IS_KUBERNETES' ? (
-      <EuiCodeBlock language="yaml" style={{ maxHeight: 300 }} fontSize="m">
-        {yaml}
-      </EuiCodeBlock>
-    ) : (
-      ''
-    );
-
-  const downloadLink =
-    isK8s === 'IS_KUBERNETES'
-      ? core.http.basePath.prepend(
-          `${agentPolicyRouteService.getK8sFullDownloadPath()}?fleetServer=${fleetServer}&enrolToken=${enrollmentAPIKey}`
-        )
-      : `https://www.elastic.co/downloads/past-releases/elastic-agent-${kibanaVersionURLString}`;
-
-  const downloadMsg =
-    isK8s === 'IS_KUBERNETES' ? (
-      <FormattedMessage
-        id="xpack.fleet.agentEnrollment.downloadManifestButtonk8s"
-        defaultMessage="Download Manifest"
-      />
-    ) : (
-      <FormattedMessage
-        id="xpack.fleet.agentEnrollment.downloadLink"
-        defaultMessage="Go to download page"
-      />
-    );
-
   return (
```

Comment:
At this point this component should be separate split in two because here we are conflating components that logically serve different needs:

- `DownloadInstructions` should contain only the instructions for the fleet server case, so the prop`hasFleetServer` can be removed altogether.
- A second component `KubernetesInstructions` (or similar name) can contain all the instructions related to kubernetes, and this would just be rendered inside `PlatformSelector` in place of the current one.